chain/chain/src/chain.rs

```diff
@@ -2969,6 +2969,13 @@ impl<'a> ChainUpdate<'a> {
             {
                 return Err(ErrorKind::InvalidFinalityInfo.into());
             }
+
+            let mut block_merkle_tree =
+                self.chain_store_update.get_block_merkle_tree(&header.prev_hash)?.clone();
```

Comment:
We store it when we process the block. We could potentially do a optimization of storing the updated one when we produce the block, but I don't see the need to do that as it might introduce unnecessary bugs.